app/forms/AdditionalConstraintsMapping.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import play.api.data.validation.Constraint
+import play.api.data.{FormError, Mapping}
+import uk.gov.voa.play.form.Condition
+
+case class ConditionalConstraint[T](shouldConstraintBeApplied: Condition, errorKey: String, constraint: (T => Boolean))
+
+/**
+  * A mapping that extends another mapping with additional constraints that are only added if their
+  * conditions are satisfied.
+  *
+  * @param baseMappings the wrapped mapping that should always be applied.
+  * @param conditionalAdditionalConstraints a Seq of additional constraints to be applied, only the first one who's
```

Comment:
Nit: the explanation is not clear. Maybe `...only **if** the first one condition...`

Reply:
I've updated it to: "a Seq of additional constraints to be applied. Only the first constraints in the sequence who's condition is satisfied is added (the others are ignored)." Any clearer?